pkg/ccloudv2/utils.go

```diff
@@ -55,6 +55,19 @@ func NewRetryableHttpClient(unsafeTrace bool) *http.Client {
 	return client.StandardClient()
 }
 
+func NewRetryableHttpClientWithRedirect(unsafeTrace bool, checkRedirect func(req *http.Request, via []*http.Request) error) *http.Client {
```

Comment:
The existing `NewRetryableHttpClient` [returns](https://github.com/confluentinc/cli/blob/6b98a5f13e36ae1171781762b3bbe9dd76b86fb1/pkg/ccloudv2/utils.go#L68) a net/http client while using the retryable http implementation internally. 

Initially, I tried using the net/http client by adding the `CheckRedirect` directly by using `cfg.HTTPClient.CheckRedirect = func(...)`. However, it did not work out as my function was never called. The redirection implementation must be set directly on the retryable http implementation. 

Just to give a little background, this change is temporary and we may move to automatic server side redirection in the future. I thought it would be best to create a new function which can be removed at the client side by switching back to the orignal `NewRetryableHttpClient`

lmk if my understanding is incorrect or there are better ways to do it.

Reply:
What's the default value of `CheckRetry`? Can we choose to always set it, and pass that default value to the `NewRetryableHttpClient()` function?

Reply:
DefaultCheckRetry is here - https://github.com/hashicorp/go-retryablehttp/blob/main/client.go#L446-L455 but we don't use it. We have our own - https://github.com/confluentinc/cli/blob/6b98a5f13e36ae1171781762b3bbe9dd76b86fb1/pkg/ccloudv2/utils.go#L61-L66
Or did you mean `CheckRedirect` ?

Reply:
Oops, yep- I meant `CheckRedirect`.

Reply:
There is no/nil `CheckRedirect` defined in retryable http implementation - https://github.com/hashicorp/go-retryablehttp/blob/main/client.go . So, I think the default would be on net/http implementation from this comment - https://github.com/confluentinc/cli/pull/2231#discussion_r1312218988 . lmk if my understanding is incorrect. 

The default `CheckRedirect` works for other confluent clients because it does not require the auth token required by the flink gateway client to work correctly. While retrying, the default net/http implementation drops Authorization header which is required for authnz in flink gateway. Other confluent clients may choose to implement it in case of redirection but that maybe beyond the scope of this PR. Also, changing the `NewRetryableHttpClient` directly would increase the blast radius of this change.